src/benchmark_mediation.py

```diff
@@ -935,6 +935,248 @@ def medDML(y, t, m, x, trim=0.05, order=1):
     return list(raw_res_R[0, :5]) + [ntrimmed]
 
 
+def med_dml(
+    x,
+    t,
+    m,
+    y,
+    crossfit=0,
+    trim=0.05,
+    normalized=True,
+    regularization=True,
+    random_state=None,
+):
+    """
+    Python implementation of Double Machine Learning procedure, as described in :
+    Helmut Farbmacher and others, Causal mediation analysis with double machine learning,
+    The Econometrics Journal, Volume 25, Issue 2, May 2022, Pages 277–300,
+    https://doi.org/10.1093/ectj/utac003
+
+    Parameters
+    ----------
+    x : array-like, shape (n_samples, n_features_covariates)
+        Covariates value for each unit
+        Can be multidimensional or continuous
+    t : array-like, shape (n_samples)
+        Treatment value for each unit
+    m : array-like, shape (n_samples, n_features_mediator)
+        Mediator value for each unit
+        Can be multidimensional or continuous
+    y : array-like, shape (n_samples)
+        Outcome value for each unit
+    crossfit : int, default=0
+        Number of folds for cross-fitting.
+    trim : float, default=0.05
+        Trimming treshold for discarding observations with extreme probability.
+    normalized : boolean, default=True
+        Normalizes the inverse probability-based weights.
+    regularization : boolean, default=True
+        Whether to use regularized models (logistic or linear regression).
+        If True, cross-validation is used to chose among 8 potential
+        log-spaced values between 1e-5 and 1e5.
+    random_state : int, default=None
+        LogisticRegression random state instance.
+
+    Returns
+    -------
+    tuple
+        A tuple of estimated effects :
```

Comment:
This is a bad pattern. We should have 6 return arguments with explicit type and semantics.